### PR TITLE
Move Noop Trace implementation to one class.

### PR DESCRIPTION
**Pull request by songy23**
Updates https://github.com/bogdandrutu/openconsensus/issues/65.

### CODE REVIEW

**Comment by songy23** [comment]
Per discussion `BlankSpan` is still under its own class since it can be used in classes other than the noop impl.

File: api/src/main/java/openconsensus/trace/NoopTrace.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      Utils.checkNotNull(name, "name");
+    }
+  }
+
+  /**
+   * Returns an instance that contains no-op implementations for all the instances.
+   *
+   * @return an instance that contains no-op implementations for all the instances.
+   */
+  private static PropagationComponent getNoopPropagationComponent() {
```
Comment (bogdandrutu) [inline api/src/main/java/openconsensus/trace/NoopTrace.java]: We should be consistent to not have all of these singletons. Use newNoopPropagationComponent and cache the result in the NoopTraceComponent

Comment (songy23) [reply api/src/main/java/openconsensus/trace/NoopTrace.java]: In `NoopTags` we use singleton:
https://github.com/bogdandrutu/openconsensus/blob/958165da6b5175d3c00f2c41aee7e03fa645bfdb/api/src/main/java/openconsensus/tags/NoopTags.java#L73.

I think it makes sense to keep consistent here.

Comment (songy23) [reply api/src/main/java/openconsensus/trace/NoopTrace.java]: https://github.com/bogdandrutu/openconsensus/pull/88/commits/d64a8d290d33ba6675daa5617ff8b06c2f34a475 removed all singleton. Will do another PR to remove it from other noop impl.

**Pull request merged**